src/cmdlib.sh

```diff
@@ -102,3 +113,81 @@ runcompose() {
          ${TREECOMPOSE_FLAGS:-} ${manifest} "$@"
     set +x
 }
+
+prepare_vm() {
+    # rpms to create appliance VM out of
+    rpms=' bash vim-minimal coreutils util-linux procps-ng kmod kernel-modules'
+    rpms+=' cifs-utils' # for samba
+    rpms+=' systemd' # for clean reboot
+    rpms+=' dhcp-client bind-export-libs iproute' # networking
+    rpms+=' rpm-ostree distribution-gpg-keys' # to run the compose
+    rpms+=' selinux-policy selinux-policy-targeted policycoreutils' #selinux
+
+    # prepare appliance VM
+    supermin -v --prepare --use-installed $rpms -o "${superminpreparedir}"
+    supermin -v --build "${superminpreparedir}" \
+             --include-packagelist --size 4G -f ext2 -o "${superminbuilddir}"
+
+    # Create a disk image for our caches (pkgcache and bare-user build repo)
+    [ ! -d "${workdir}/tmp/emptydir" ] && mkdir "${workdir}/tmp/emptydir"
+    if [ ! -f "${cachesimg}" ]; then
+        virt-make-fs --format=qcow2 --type=xfs \
+                     --size=10G "${workdir}/tmp/emptydir" "${cachesimg}"
+    fi
+}
+
+run_vm() {
+    init=$1
+    set -x
+
+    # Copy in the init script with the code we want to run  
+    chmod +x "${init}"
+    virt-copy-in -a "${superminbuilddir}/root" "${init}" /
+
+    # Execute unprivileged VM to run compose. Some notes:
+    # - smb="${workdir}" - we'll serve our workdir over samba
+    # - sda - rootfs from supermin build
+    # - sdb - caches.qcow2 - we store the bare-user repo and pkgcache here
+    # - discard=unmap - using virtio-scsi disks and discard=unmap so we can fstrim
+    #                   and recover disk space from the VM
+    qemu-kvm -nodefaults -nographic -m 2048 -no-reboot \
+              -kernel "${superminbuilddir}/kernel" \
+              -initrd "${superminbuilddir}/initrd" \
+              -netdev user,id=eth0,hostname=supermin,smb="${workdir}",hostfwd=tcp:127.0.0.1:8000-:8000 \
```

Comment:
Since today we run with `--net=host`, suddenly we'll need to worry about port conflicts.  It is tempting to go back to the default bridged networking.  I forget if there was a reason I added `--net=host` to the default arguments...oh oh right, it's because I use `dns=dnsmasq` on my host and podman today can't handle that.  Urgh.  I can deal with working around that here.

But in general...for sharing content between the container and VM I lean a bit towards using say rsync-over-ssh-over-virtio or something like that - there's no good reason to use TCP here.

Reply:
actually I can remove the `hostfwd` I don't actually need it. I was experimenting to see if pulling via HTTP was any faster than doing the `ostree pull-local` over smb. it wasn't, so I'll just remove this.

> Since today we run with `--net=host`

I've actually been running without `--net=host` in my tests, so my plan was to actually remove that. Are you saying that would be problematic?

Reply:
It's OK to remove if that turns out to be useful, I'll figure out how to make my current setup work with default bridged.